### ISSUE TITLE
Exception on starting test in os x 10.11.1 (15B42)

### ISSUE THREAD

**Issue by ojosdegris** [lefthandedgoat/canopy#259]
Checkouted canopyStarterKit, it works fine on windows, but when I am running `sh .\build.sh` or `sh .\build.sh All` in os x I am getting exception (chrome window got opened with `data:,` in the url): 

```
  mono  bin/canopyStarterKit/canopyStarterKit.exe --browser Chrome --tag All --testtype UnderDevelopment
  Starting ChromeDriver 2.21.371459 (36d3d07f660ff2bc1bf28a75d1cdabed0983e7c4) on port 50756 
  Only local connections are allowed.
  Stacktrace:

  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Windows.Forms.XplatUICarbon.CGDisplayBounds (intptr) <0xffffffff>
  at System.Windows.Forms.XplatUICarbon.get_WorkingArea () <0x00035>
  at System.Windows.Forms.XplatUICarbon.get_VirtualScreen () <0x00020>
  at System.Windows.Forms.XplatUI.get_VirtualScreen () <0x00025>
  at System.Windows.Forms.Screen..cctor () <0x0011f>
  at (wrapper runtime-invoke) object.runtime_invoke_void (object,intptr,intptr,intptr) <0xffffffff>
  at <unknown> <0xffffffff>
  at canopy.core.pin (canopy.types/direction) <0x00028>
  at canopy.core.start (canopy.types/BrowserStartMode) <0x01466>
  at Program.main (string[]) <0x00061>
  at (wrapper runtime-invoke) <Module>.runtime_invoke_int_object (object,intptr,intptr,intptr) <0xffffffff>


  Native stacktrace:


  Debug info from gdb:

  (lldb) command source -s 0 '/tmp/mono-gdb-commands.j7So8I'
  Executing commands in '/tmp/mono-gdb-commands.j7So8I'.
  (lldb) process attach --pid 97446
  Process 97446 stopped
  * thread #1: tid = 0xe86850, 0x00007fff9655072a libsystem_kernel.dylib`__wait4 + 10, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
      frame #0: 0x00007fff9655072a libsystem_kernel.dylib`__wait4 + 10
  libsystem_kernel.dylib`__wait4:
  ->  0x7fff9655072a <+10>: jae    0x7fff96550734            ; <+20>
      0x7fff9655072c <+12>: movq   %rax, %rdi
      0x7fff9655072f <+15>: jmp    0x7fff9654b414            ; cerror
      0x7fff96550734 <+20>: retq   

  Executable module set to "/usr/local/bin/mono".
  Architecture set to: x86_64h-apple-macosx.
  (lldb) thread list
  Process 97446 stopped
  * thread #1: tid = 0xe86850, 0x00007fff9655072a libsystem_kernel.dylib`__wait4 + 10, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
    thread #2: tid = 0xe86851, 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
    thread #3: tid = 0xe86852, 0x00007fff9654acd2 libsystem_kernel.dylib`semaphore_wait_trap + 10
    thread #4: tid = 0xe86853, 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
    thread #5: tid = 0xe86854, 0x00007fff965510a2 libsystem_kernel.dylib`kevent_qos + 10, queue = 'com.apple.libdispatch-manager'
    thread #6: tid = 0xe86855, 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
    thread #7: tid = 0xe86856, 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
    thread #8: tid = 0xe8685d, 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
    thread #9: tid = 0xe8685e, 0x00007fff965502b2 libsystem_kernel.dylib`__semwait_signal + 10
    thread #10: tid = 0xe8685f, 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
    thread #11: tid = 0xe86860, 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
    thread #12: tid = 0xe86861, 0x00007fff96550222 libsystem_kernel.dylib`__select + 10
    thread #13: tid = 0xe86862, 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
    thread #14: tid = 0xe86937, 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
  (lldb) thread backtrace all
  * thread #1: tid = 0xe86850, 0x00007fff9655072a libsystem_kernel.dylib`__wait4 + 10, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
    * frame #0: 0x00007fff9655072a libsystem_kernel.dylib`__wait4 + 10
      frame #1: 0x000000010fb39a0b mono`mono_handle_native_sigsegv + 421
      frame #2: 0x000000010fb8815c mono`mono_arch_handle_altstack_exception + 90
      frame #3: 0x000000010faaa373 mono`mono_sigsegv_signal_handler + 430
      frame #4: 0x00007fff9b01252a libsystem_platform.dylib`_sigtramp + 26
      frame #5: 0x00007fff8c3f7e69 CoreGraphics`CGDisplayBounds + 48
      frame #6: 0x000000011435697f
      frame #7: 0x000000011011c30f
      frame #8: 0x000000010faac8ca mono`mono_jit_runtime_invoke + 1463
      frame #9: 0x000000010fc29c97 mono`mono_runtime_invoke + 110
      frame #10: 0x000000010fc2a159 mono`mono_runtime_class_init_full + 775
      frame #11: 0x000000010faa78ba mono`mono_jit_compile_method_inner + 2004
      frame #12: 0x000000010faa9dbd mono`mono_jit_compile_method_with_opt + 504
      frame #13: 0x000000010faa9b84 mono`mono_jit_compile_method + 42
      frame #14: 0x000000010fb3b238 mono`common_call_trampoline + 1180
      frame #15: 0x000000011005b189
      frame #16: 0x0000000113fc7969
      frame #17: 0x000000010faac8ca mono`mono_jit_runtime_invoke + 1463
      frame #18: 0x000000010fc29c97 mono`mono_runtime_invoke + 110
      frame #19: 0x000000010fc2e915 mono`mono_runtime_exec_main + 379
      frame #20: 0x000000010fb0f800 mono`mono_main + 6785
      frame #21: 0x00007fff9a64a5ad libdyld.dylib`start + 1

    thread #2: tid = 0xe86851, 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
      frame #0: 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
      frame #1: 0x00007fff9b8d873d libsystem_pthread.dylib`_pthread_cond_wait + 767
      frame #2: 0x000000010fc74811 mono`thread_func + 273
      frame #3: 0x00007fff9b8d79b1 libsystem_pthread.dylib`_pthread_body + 131
      frame #4: 0x00007fff9b8d792e libsystem_pthread.dylib`_pthread_start + 168
      frame #5: 0x00007fff9b8d5385 libsystem_pthread.dylib`thread_start + 13

    thread #3: tid = 0xe86852, 0x00007fff9654acd2 libsystem_kernel.dylib`semaphore_wait_trap + 10
      frame #0: 0x00007fff9654acd2 libsystem_kernel.dylib`semaphore_wait_trap + 10
      frame #1: 0x000000010fc913dc mono`mono_sem_wait + 29
      frame #2: 0x000000010fc27fa4 mono`finalizer_thread + 124
      frame #3: 0x000000010fc09ed1 mono`start_wrapper + 410
      frame #4: 0x000000010fc97bf9 mono`inner_start_thread + 197
      frame #5: 0x00007fff9b8d79b1 libsystem_pthread.dylib`_pthread_body + 131
      frame #6: 0x00007fff9b8d792e libsystem_pthread.dylib`_pthread_start + 168
      frame #7: 0x00007fff9b8d5385 libsystem_pthread.dylib`thread_start + 13

    thread #4: tid = 0xe86853, 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
      frame #0: 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
      frame #1: 0x00007fff9b8d758c libsystem_pthread.dylib`_pthread_wqthread + 1283
      frame #2: 0x00007fff9b8d5375 libsystem_pthread.dylib`start_wqthread + 13

    thread #5: tid = 0xe86854, 0x00007fff965510a2 libsystem_kernel.dylib`kevent_qos + 10, queue = 'com.apple.libdispatch-manager'
      frame #0: 0x00007fff965510a2 libsystem_kernel.dylib`kevent_qos + 10
      frame #1: 0x00007fff89ac211d libdispatch.dylib`_dispatch_mgr_invoke + 216
      frame #2: 0x00007fff89ac1d85 libdispatch.dylib`_dispatch_mgr_thread + 52

    thread #6: tid = 0xe86855, 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
      frame #0: 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
      frame #1: 0x00007fff9b8d758c libsystem_pthread.dylib`_pthread_wqthread + 1283
      frame #2: 0x00007fff9b8d5375 libsystem_pthread.dylib`start_wqthread + 13

    thread #7: tid = 0xe86856, 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
      frame #0: 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
      frame #1: 0x00007fff9b8d758c libsystem_pthread.dylib`_pthread_wqthread + 1283
      frame #2: 0x00007fff9b8d5375 libsystem_pthread.dylib`start_wqthread + 13

    thread #8: tid = 0xe8685d, 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
      frame #0: 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
      frame #1: 0x00007fff9b8d873d libsystem_pthread.dylib`_pthread_cond_wait + 767
      frame #2: 0x000000010fc792e4 mono`_wapi_handle_timedwait_signal_handle + 486
      frame #3: 0x000000010fc867ff mono`wapi_WaitForSingleObjectEx + 502
      frame #4: 0x000000010fc06b71 mono`mono_wait_uninterrupted + 112
      frame #5: 0x000000010fc06d48 mono`ves_icall_System_Threading_WaitHandle_WaitOne_internal + 78
      frame #6: 0x0000000113c21654
      frame #7: 0x0000000111ec447c mscorlib.dll.dylib`System_Threading_WaitHandle_WaitOne_int + 28
      frame #8: 0x0000000111ec3813 mscorlib.dll.dylib`System_Threading_Timer_Scheduler_SchedulerThread + 2467
      frame #9: 0x0000000111d1f48b mscorlib.dll.dylib`System_Threading_ThreadHelper_ThreadStart_Context_object + 155
      frame #10: 0x0000000111d1dbc1 mscorlib.dll.dylib`System_Threading_ExecutionContext_Run_System_Threading_ExecutionContext_System_Threading_ContextCallback_object_bool + 33
      frame #11: 0x0000000111d1db4a mscorlib.dll.dylib`System_Threading_ExecutionContext_Run_System_Threading_ExecutionContext_System_Threading_ContextCallback_object + 90
      frame #12: 0x0000000111d1f5df mscorlib.dll.dylib`System_Threading_ThreadHelper_ThreadStart + 47
      frame #13: 0x000000011011dfb1
      frame #14: 0x000000010faac8ca mono`mono_jit_runtime_invoke + 1463
      frame #15: 0x000000010fc29c97 mono`mono_runtime_invoke + 110
      frame #16: 0x000000010fc09f0c mono`start_wrapper + 469
      frame #17: 0x000000010fc97bf9 mono`inner_start_thread + 197
      frame #18: 0x00007fff9b8d79b1 libsystem_pthread.dylib`_pthread_body + 131
      frame #19: 0x00007fff9b8d792e libsystem_pthread.dylib`_pthread_start + 168
      frame #20: 0x00007fff9b8d5385 libsystem_pthread.dylib`thread_start + 13

    thread #9: tid = 0xe8685e, 0x00007fff965502b2 libsystem_kernel.dylib`__semwait_signal + 10
      frame #0: 0x00007fff965502b2 libsystem_kernel.dylib`__semwait_signal + 10
      frame #1: 0x00007fff8c39ca75 libsystem_c.dylib`nanosleep + 199
      frame #2: 0x000000010fc88089 mono`wapi_SleepEx + 149
      frame #3: 0x000000010fc0c5d3 mono`monitor_thread + 142
      frame #4: 0x000000010fc09ed1 mono`start_wrapper + 410
      frame #5: 0x000000010fc97bf9 mono`inner_start_thread + 197
      frame #6: 0x00007fff9b8d79b1 libsystem_pthread.dylib`_pthread_body + 131
      frame #7: 0x00007fff9b8d792e libsystem_pthread.dylib`_pthread_start + 168
      frame #8: 0x00007fff9b8d5385 libsystem_pthread.dylib`thread_start + 13

    thread #10: tid = 0xe8685f, 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
      frame #0: 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
      frame #1: 0x00007fff9b8d873d libsystem_pthread.dylib`_pthread_cond_wait + 767
      frame #2: 0x000000010fc8ef20 mono`mono_cond_timedwait_ms + 141
      frame #3: 0x000000010fc0ccdb mono`worker_thread + 776
      frame #4: 0x000000010fc09ed1 mono`start_wrapper + 410
      frame #5: 0x000000010fc97bf9 mono`inner_start_thread + 197
      frame #6: 0x00007fff9b8d79b1 libsystem_pthread.dylib`_pthread_body + 131
      frame #7: 0x00007fff9b8d792e libsystem_pthread.dylib`_pthread_start + 168
      frame #8: 0x00007fff9b8d5385 libsystem_pthread.dylib`thread_start + 13

    thread #11: tid = 0xe86860, 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
      frame #0: 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
      frame #1: 0x00007fff9b8d873d libsystem_pthread.dylib`_pthread_cond_wait + 767
      frame #2: 0x000000010fc8ef20 mono`mono_cond_timedwait_ms + 141
      frame #3: 0x000000010fc0ccdb mono`worker_thread + 776
      frame #4: 0x000000010fc09ed1 mono`start_wrapper + 410
      frame #5: 0x000000010fc97bf9 mono`inner_start_thread + 197
      frame #6: 0x00007fff9b8d79b1 libsystem_pthread.dylib`_pthread_body + 131
      frame #7: 0x00007fff9b8d792e libsystem_pthread.dylib`_pthread_start + 168
      frame #8: 0x00007fff9b8d5385 libsystem_pthread.dylib`thread_start + 13

    thread #12: tid = 0xe86861, 0x00007fff96550222 libsystem_kernel.dylib`__select + 10
      frame #0: 0x00007fff96550222 libsystem_kernel.dylib`__select + 10
      frame #1: 0x000000010fc90e17 mono`mono_poll + 387
      frame #2: 0x000000010fc0e344 mono`poll_event_wait + 88
      frame #3: 0x000000010fc0dee1 mono`selector_thread + 933
      frame #4: 0x000000010fc09ed1 mono`start_wrapper + 410
      frame #5: 0x000000010fc97bf9 mono`inner_start_thread + 197
      frame #6: 0x00007fff9b8d79b1 libsystem_pthread.dylib`_pthread_body + 131
      frame #7: 0x00007fff9b8d792e libsystem_pthread.dylib`_pthread_start + 168
      frame #8: 0x00007fff9b8d5385 libsystem_pthread.dylib`thread_start + 13

    thread #13: tid = 0xe86862, 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
      frame #0: 0x00007fff9654ff5e libsystem_kernel.dylib`__psynch_cvwait + 10
      frame #1: 0x00007fff9b8d873d libsystem_pthread.dylib`_pthread_cond_wait + 767
      frame #2: 0x000000010fc8ef20 mono`mono_cond_timedwait_ms + 141
      frame #3: 0x000000010fc0ccdb mono`worker_thread + 776
      frame #4: 0x000000010fc09ed1 mono`start_wrapper + 410
      frame #5: 0x000000010fc97bf9 mono`inner_start_thread + 197
      frame #6: 0x00007fff9b8d79b1 libsystem_pthread.dylib`_pthread_body + 131
      frame #7: 0x00007fff9b8d792e libsystem_pthread.dylib`_pthread_start + 168
      frame #8: 0x00007fff9b8d5385 libsystem_pthread.dylib`thread_start + 13

    thread #14: tid = 0xe86937, 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
      frame #0: 0x00007fff9655078a libsystem_kernel.dylib`__workq_kernreturn + 10
      frame #1: 0x00007fff9b8d758c libsystem_pthread.dylib`_pthread_wqthread + 1283
      frame #2: 0x00007fff9b8d5375 libsystem_pthread.dylib`start_wqthread + 13
  (lldb) detach

  =================================================================
  Got a SIGSEGV while executing native code. This usually indicates
  a fatal error in the mono runtime or one of the native libraries 
  used by your application.
  =================================================================

  Process 97446 detached
  Running build failed.
  Error:
  System.Exception: Failed result from unit tests
    at FSI_0005.Build.run (System.String args) <0x10f640720 + 0x00117> in <filename unknown>:0 
    at FSI_0005.Build+clo@124-9.Invoke (Microsoft.FSharp.Core.Unit _arg1) <0x10f6406f0 + 0x0001a> in <filename unknown>:0 
    at Fake.TargetHelper+targetFromTemplate@195[a].Invoke (Microsoft.FSharp.Core.Unit unitVar0) <0x10f624aa0 + 0x00023> in <filename unknown>:0 
    at Fake.TargetHelper.runSingleTarget (Fake.TargetTemplate`1 target) <0x10f619150 + 0x000fd> in <filename unknown>:0 
  Status:        Failure
    1) System.Exception: Failed result from unit tests
    at FSI_0005.Build.run (System.String args) <0x10f640720 + 0x00117> in <filename unknown>:0 
    at FSI_0005.Build+clo@124-9.Invoke (Microsoft.FSharp.Core.Unit _arg1) <0x10f6406f0 + 0x0001a> in <filename unknown>:0 
    at Fake.TargetHelper+targetFromTemplate@195[a].Invoke (Microsoft.FSharp.Core.Unit unitVar0) <0x10f624aa0 + 0x00023> in <filename unknown>:0 
    at Fake.TargetHelper.runSingleTarget (Fake.TargetTemplate`1 target) <0x10f619150 + 0x000fd> in <filename unknown>:0 

  (lldb) quit
```

Tweaked `build.fsx` to start Safari instead of Chrome and getting this:

```
  mono  bin/canopyStarterKit/canopyStarterKit.exe --browser Safari --tag All --testtype All
  2016-04-25 03:34:20.004 Safari[97836:15239408] Connection error while checking Apple Internalness. Error: Error Domain=NSCocoaErrorDomain Code=4097 "connection to service named com.apple.CrashReporterSupportHelper" UserInfo={NSDebugDescription=connection to service named com.apple.CrashReporterSupportHelper}
```

Safari opens

**Comment by lefthandedgoat** [comment]
Can you tell me what version of Mono you have installed on OSX?  I just did a fresh clone and it works for me with this version of mono:

```
mono -V
Mono JIT compiler version 4.4.0 (mono-4.4.0-branch/a3fabf1 Fri Apr  8 13:48:29 EDT 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com

```

**Comment by lefthandedgoat** [comment]
You can also add this before the line that starts the browser:

```canopy.configuration. autoPinBrowserRightOnLaunch <- false ```

https://github.com/lefthandedgoat/canopy/blob/master/src/canopy/configuration.fs#L29

It looks like its failing when trying to resize the window at startup.

**Comment by ojosdegris** [comment]
Hi,
here you go:

    $ mono -V                                                                                                      

	Mono JIT compiler version 4.2.3 (Stable 4.2.3.4/832de4b Wed Mar 30 13:57:48 PDT 2016)
	Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
		TLS:           normal
		SIGSEGV:       altstack
		Notification:  kqueue
		Architecture:  amd64
		Disabled:      none
		Misc:          softdebug
		LLVM:          supported, not enabled.
		GC:            sgen

**Comment by ojosdegris** [comment]
configuration.autoPinBrowserRightOnLaunch <- false

in the Program.main actually did it. Thanks!

**Comment by ojosdegris** [comment]
One thing that I wanted to ask you while you looking at it, how to update html report template. I basically have 2 ideas:
 
   - change report to xml and make it compatible with junit, so other CI tools like Bamboo can use it ( we are using bamboo)

   - make report use react.js as it can be written in more functional way, basically map test results to stateless functions ( that works starting react.js v0.15 ) like that:

        const ReportLine = (props) => {
           const { testName, result, time } = props
           return (<div>
               <span className='name'>{testName}</span>
               <span className='result'>{result}</span>
               <span className='time'>{time}</span>
           </div>)
        }

        const Report  = (props) => {
            const {tests , context } = props
            return (<div>
                 <h1>{context}</h1>
                {tests.map((t) => ReportLine(t))}
          </div>)
        }

**Comment by lefthandedgoat** [comment]
There are currently 3 reporters, the console, html, and one for team city.

For bamboo, if it hooks into the console like team city does, you can just make a special console reporter that is bamboo aware.  Like this: https://github.com/lefthandedgoat/canopy/blob/master/src/canopy/reporters.fs#L111-L179

If it does not, the xml path could be a way to go about it.  Right now I don't actually store results in a collection to them serialize to xml or json but that is a possibility.  I don't think it would be too much work to make that work.

I could push data into a collection that is watched by react, but at the moment I do not know enough react.

Another option for bamboo (this works in jenkins) is to take the html that is generated and save it as an artifact.  canopy will grab all the conents of the report and save it, and you just need your CI tool to 'attach' it to your run. You have to set the report path https://github.com/lefthandedgoat/canopy/blob/master/src/canopy/reporters.fs#L228 with `Some "c:\whatever"` then your tool can pick up that artifact.

Hope this helps, thanks for the feedback!

**Comment by ojosdegris** [comment]
If you help me to figure out how to better add another parameter to livehtmlreporter to disable pin, the same way it is done for configuration , i can do pull request =)

**Comment by lefthandedgoat** [comment]
I think it will have to be an option/defaulted parameter for the constructor:

https://github.com/lefthandedgoat/canopy/blob/master/src/canopy/reporters.fs#L181

you can read about optional parameters here: https://msdn.microsoft.com/en-us/library/dd233213.aspx

Then you can do a check to make sure that you should run the pin code in here:

https://github.com/lefthandedgoat/canopy/blob/master/src/canopy/reporters.fs#L206

**Comment by ojosdegris** [comment]
Thanks! let me try to do that.

**Comment by ojosdegris** [comment]
If you will have time to go with xml path, ping me I'd love to do it but again will probably need some guidance since I am really new to f#

**Comment by ojosdegris** [comment]
Also, just confirmed, after updating mono to 4.4 alpha exception goes away with livehtmlreporter

**Comment by lefthandedgoat** [comment]
Ok great news on mono.  I used 4.2.x briefly and never had problems, so maybe one of the minor versions had a bug.

I accepted your changes and tested them and they look good.

I am going to make a new branch and do the basics of the xml thing.  Then you can pick up from there and make it conform to the junit structure (never used junit before).

**Comment by ojosdegris** [comment]
Thank you!

Once we do that we can pretty much stick canopy everywhere where junit xml is accepted.

People seems to use xsd from this repo https://github.com/windyroad/JUnit-Schema and I can do mapping once I have collection of tests run. We probably need another issue for that...

**Comment by lefthandedgoat** [comment]
xmlReporter branch has a working reporter (already turned on), it writes to c:\canopyresults.xml

https://github.com/lefthandedgoat/canopy/commit/d664d09bb3c3091b86c6adce66cc38fca0f050aa

I dont know a lot about classes and OOP in F#, and I wanted it to interop with XML serializer, so its not really idiomatic f#, but it works.  I used normal classes and List<T> from C#

hope this helps

**Comment by ojosdegris** [comment]
Thank you Chriss, let me dig into it now

**Comment by baronfel** [comment]
This report is very similar to my own.  

OS: OSX 10.11.6
Driver: Chromedriver 2.24.417412 (ac882d3ce7c0d99292439bf3405780058fcca0a6)
Mono: 
```
Mono JIT compiler version 4.6.0 (Stable 4.6.0.245/746756c Mon Sep 19 13:57:55 BST 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	LLVM:          supported, not enabled.
	GC:            sgen
```

Source: https://github.com/baronfel/fsharp-lang-suggestions, branch canopy_test_osx
Command: ./run.sh

Output: 
```
✘  ~/code/fsharp-lang-suggestions   canopy_test_osx ●  (lldb) ./run.sh
Checking Paket version (downloading latest stable)...
Paket.exe 3.23.2 is up to date.
Paket version 3.23.2.0
0 seconds - ready.
Microsoft (R) Build Engine version 14.1.0.0
Copyright (C) Microsoft Corporation. All rights reserved.

Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
Build started 10/11/2016 8:54:35 AM.
Project "/Users/chet/code/fsharp-lang-suggestions/fslang-migration.sln" on node 1 (default targets).
ValidateSolutionConfiguration:
  Building solution configuration "Release|Any CPU".
Project "/Users/chet/code/fsharp-lang-suggestions/fslang-migration.sln" (1) is building "/Users/chet/code/fsharp-lang-suggestions/src/fslang-migration.fsproj" (2) on node 1 (default targets).
CoreCompile:
  Tool /Library/Frameworks/Mono.framework/Versions/4.6.1/lib/mono/4.5/fsc.exe execution started with arguments: -o:obj/Release/fslang_migration.exe --debug:pdbonly --noframework --define:TRACE --doc:bin/Release/fslang_migration.XML --optimize+ --platform:anycpu -r:/Users/chet/code/fsharp-lang-suggestions/packages/canopy/lib/canopy.dll -r:/Users/chet/code/fsharp-lang-suggestions/packages/DotLiquid/lib/net451/DotLiquid.dll -r:/Library/Frameworks/Mono.framework/Versions/4.6.1/lib/mono/4.5/FSharp.Core.dll -r:/Users/chet/code/fsharp-lang-suggestions/packages/FSharp.Data/lib/net40/FSharp.Data.dll -r:/Library/Frameworks/Mono.framework/Versions/4.6.1/lib/mono/4.5/mscorlib.dll -r:/Users/chet/code/fsharp-lang-suggestions/packages/Newtonsoft.Json/lib/net45/Newtonsoft.Json.dll -r:/Users/chet/code/fsharp-lang-suggestions/packages/Octokit/lib/net45/Octokit.dll -r:/Users/chet/code/fsharp-lang-suggestions/packages/RestSharp/lib/net46/RestSharp.dll -r:/Library/Frameworks/Mono.framework/Versions/4.6.1/lib/mono/4.5/System.Core.dll -r:/Library/Frameworks/Mono.framework/Versions/4.6.1/lib/mono/4.5/System.dll -r:/Library/Frameworks/Mono.framework/Versions/4.6.1/lib/mono/4.5/System.Drawing.dll -r:/Library/Frameworks/Mono.framework/Versions/4.6.1/lib/mono/4.5/System.Net.Http.dll -r:/Library/Frameworks/Mono.framework/Versions/4.6.1/lib/mono/4.5/System.Numerics.dll -r:/Library/Frameworks/Mono.framework/Versions/4.6.1/lib/mono/4.5/System.Xml.Linq.dll -r:/Users/chet/code/fsharp-lang-suggestions/packages/UserVoice/lib/UserVoice.dll -r:/Users/chet/code/fsharp-lang-suggestions/packages/Selenium.WebDriver/lib/net40/WebDriver.dll --target:exe --warn:3 --warnaserror:76 --fullpaths --flaterrors --highentropyva- Input.fs Model.fs Api.fs Templating.fs Scraping.fs Github.fs Program.fs
  F# Compiler for F# 4.1
  Freely distributed under the Apache 2.0 Open Source License
_CopyAppConfigFile:
  Copying file from "obj/Release/fslang-migration.fsproj.fslang_migration.exe.config" to "bin/Release/fslang_migration.exe.config".
CopyFilesToOutputDirectory:
  Copying file from "obj/Release/fslang_migration.exe" to "bin/Release/fslang_migration.exe".
  fslang-migration -> /Users/chet/code/fsharp-lang-suggestions/src/bin/Release/fslang_migration.exe
Done Building Project "/Users/chet/code/fsharp-lang-suggestions/src/fslang-migration.fsproj" (default targets).
Done Building Project "/Users/chet/code/fsharp-lang-suggestions/fslang-migration.sln" (default targets).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:05.92
Starting ChromeDriver 2.24.417412 (ac882d3ce7c0d99292439bf3405780058fcca0a6) on port 53095
Only local connections are allowed.
WARNING: The Carbon driver has not been ported to 64bits, and very few parts of Windows.Forms will work properly, or at all
Stacktrace:

  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Windows.Forms.XplatUICarbon.CGDisplayBounds (intptr) <0x0006e>
  at System.Windows.Forms.XplatUICarbon.get_WorkingArea () <0x00035>
  at System.Windows.Forms.XplatUICarbon.get_VirtualScreen () <0x00020>
  at System.Windows.Forms.XplatUI.get_VirtualScreen () <0x00025>
  at System.Windows.Forms.Screen..cctor () <0x0011f>
  at (wrapper runtime-invoke) object.runtime_invoke_void (object,intptr,intptr,intptr) <0x0006e>
  at <unknown> <0xffffffff>
  at canopy.core.pin (canopy.types/direction) <0x00028>
  at canopy.core.start (canopy.types/BrowserStartMode) <0x01ab8>
  at FslangMigration.Scrape.scrapeData (string) <0x00024>
  at <StartupCode$FSI_0002>.$FSI_0002.main@ () <0x0006d>
  at (wrapper runtime-invoke) object.runtime_invoke_void (object,intptr,intptr,intptr) <0x000d3>
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&) <0x00073>
  at System.Reflection.MonoMethod.Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo) <0x000a3>
  at System.RuntimeType.InvokeMember (string,System.Reflection.BindingFlags,System.Reflection.Binder,object,object[],System.Reflection.ParameterModifier[],System.Globalization.CultureInfo,string[]) <0x00eb5>
  at System.Reflection.Emit.TypeBuilder.InvokeMember (string,System.Reflection.BindingFlags,System.Reflection.Binder,object,object[],System.Reflection.ParameterModifier[],System.Globalization.CultureInfo,string[]) <0x00069>
  at System.Type.InvokeMember (string,System.Reflection.BindingFlags,System.Reflection.Binder,object,object[],System.Globalization.CultureInfo) <0x0005e>
  at Microsoft.FSharp.Compiler.AbstractIL.ILRuntimeWriter/execEntryPtFun@2017-1.Invoke (Microsoft.FSharp.Core.Unit) <0x000a8>
  at Microsoft.FSharp.Compiler.Interactive.Shell/clo@895-37.Invoke (Microsoft.FSharp.Core.FSharpFunc`2<Microsoft.FSharp.Core.Unit, Microsoft.FSharp.Core.FSharpOption`1<System.Exception>>) <0x00022>
  at Microsoft.FSharp.Primitives.Basics.List.iter<T_REF> (Microsoft.FSharp.Core.FSharpFunc`2<T_REF, Microsoft.FSharp.Core.Unit>,Microsoft.FSharp.Collections.FSharpList`1<T_REF>) <0x00047>
  at Microsoft.FSharp.Collections.ListModule.Iterate<T_REF> (Microsoft.FSharp.Core.FSharpFunc`2<T_REF, Microsoft.FSharp.Core.Unit>,Microsoft.FSharp.Collections.FSharpList`1<T_REF>) <0x0003c>
  at Microsoft.FSharp.Compiler.Interactive.Shell.arg10@894 (Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompiler,Microsoft.FSharp.Collections.FSharpList`1<Microsoft.FSharp.Core.FSharpFunc`2<Microsoft.FSharp.Core.Unit, Microsoft.FSharp.Core.FSharpOption`1<System.Exception>>>,Microsoft.FSharp.Core.Unit) <0x0007d>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompiler.ProcessInputs (Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState,Microsoft.FSharp.Collections.FSharpList`1<Microsoft.FSharp.Compiler.Ast/ParsedInput>,bool,bool,bool,Microsoft.FSharp.Collections.FSharpList`1<Microsoft.FSharp.Compiler.Ast/Ident>) <0x00865>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompiler.EvalParsedDefinitions (Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState,bool,bool,Microsoft.FSharp.Collections.FSharpList`1<Microsoft.FSharp.Compiler.Ast/SynModuleDecl>) <0x002d1>
  at Microsoft.FSharp.Compiler.Interactive.Shell/clo@1644-45.Invoke (Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState) <0x01b5e>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor.InteractiveCatch<a_REF> (Microsoft.FSharp.Core.FSharpFunc`2<a_REF, System.Tuple`2<a_REF, Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionStepStatus>>,a_REF) <0x00049>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor.ExecInteraction (bool,Microsoft.FSharp.Compiler.CompileOps/TcConfig,Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState,Microsoft.FSharp.Compiler.Ast/ParsedFsiInteraction) <0x00088>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor.ExecInteractions (bool,Microsoft.FSharp.Compiler.CompileOps/TcConfig,Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState,Microsoft.FSharp.Core.FSharpOption`1<Microsoft.FSharp.Compiler.Ast/ParsedFsiInteraction>) <0x00bd8>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor.MainThreadProcessParsedInteraction (bool,Microsoft.FSharp.Core.FSharpOption`1<Microsoft.FSharp.Compiler.Ast/ParsedFsiInteraction>,Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState) <0x001c6>
  at Microsoft.FSharp.Compiler.Interactive.Shell/res@1861.Invoke (Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState) <0x00034>
  at Microsoft.FSharp.Compiler.Interactive.Shell/run@1878-1.Invoke (Microsoft.FSharp.Core.FSharpFunc`2<Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState, System.Tuple`2<Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState, Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionStepStatus>>,Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState) <0x0001d>
  at Microsoft.FSharp.Core.OptimizedClosures/Invoke@3261<T2_REF, TResult_REF, T1_REF>.Invoke (T2_REF) <0x00029>
  at Microsoft.FSharp.Compiler.Interactive.Shell/ParseAndProcessAndEvalOneInteractionFromLexbuf@1850.Invoke (Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState) <0x00214>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor.InteractiveCatch<a_REF> (Microsoft.FSharp.Core.FSharpFunc`2<a_REF, System.Tuple`2<a_REF, Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionStepStatus>>,a_REF) <0x00049>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor.ParseAndProcessAndEvalOneInteractionFromLexbuf (bool,Microsoft.FSharp.Core.FSharpFunc`2<Microsoft.FSharp.Core.FSharpFunc`2<Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState, System.Tuple`2<Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState, Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionStepStatus>>, Microsoft.FSharp.Core.FSharpFunc`2<Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState, System.Tuple`2<Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState, Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionStepStatus>>>,Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState,Microsoft.FSharp.Compiler.LexFilter/LexFilter) <0x001b5>
  at Microsoft.FSharp.Compiler.Interactive.Shell.run@1877 (Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor,bool,Microsoft.FSharp.Compiler.LexFilter/LexFilter,Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState) <0x0007b>
  at Microsoft.FSharp.Compiler.Interactive.Shell/EvalIncludedScript@1873.Invoke (Microsoft.FSharp.Core.Unit) <0x00054>
  at Microsoft.FSharp.Compiler.Interactive.Shell.WithImplicitHome<a_REF> (Microsoft.FSharp.Compiler.CompileOps/TcConfigBuilder,string,Microsoft.FSharp.Core.FSharpFunc`2<Microsoft.FSharp.Core.Unit, a_REF>) <0x00057>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor.EvalIncludedScript (bool,Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState,string,Microsoft.FSharp.Compiler.Range/range) <0x000ea>
  at Microsoft.FSharp.Compiler.Interactive.Shell/EvalIncludedScripts@1897.Invoke (Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState) <0x00049>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor.InteractiveCatch<a_REF> (Microsoft.FSharp.Core.FSharpFunc`2<a_REF, System.Tuple`2<a_REF, Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionStepStatus>>,a_REF) <0x00049>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor.EvalIncludedScripts (Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState,bool,Microsoft.FSharp.Collections.FSharpList`1<string>) <0x000aa>
  at Microsoft.FSharp.Compiler.Interactive.Shell.consume@1907 (Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor,bool,Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState,Microsoft.FSharp.Collections.FSharpList`1<System.Tuple`2<string, bool>>) <0x0011d>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiInteractionProcessor.LoadInitialFiles (bool,Microsoft.FSharp.Compiler.Interactive.Shell/FsiDynamicCompilerState) <0x00037>
  at Microsoft.FSharp.Compiler.Interactive.Shell/FsiEvaluationSession.Run () <0x0103e>
  at Microsoft.FSharp.Compiler.Interactive.Shell.evaluateSession@2518 (string[],Microsoft.FSharp.Core.Unit) <0x000dd>
  at Microsoft.FSharp.Compiler.Interactive.Shell.MainMain (string[]) <0x001f5>
  at Microsoft.FSharp.Compiler.Interactive.Main.FsiMain (string[]) <0x00014>
  at (wrapper runtime-invoke) <Module>.runtime_invoke_int_object (object,intptr,intptr,intptr) <0x00112>

Native stacktrace:


Debug info from gdb:

(lldb) command source -s 0 '/tmp/mono-gdb-commands.QmXjOh'
Executing commands in '/tmp/mono-gdb-commands.QmXjOh'.
(lldb) process attach --pid 52756
Process 52756 stopped
* thread #1: tid = 0xfa760, 0x00007fff8f69e582 libsystem_kernel.dylib`__wait4 + 10, name = 'tid_a0b', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
    frame #0: 0x00007fff8f69e582 libsystem_kernel.dylib`__wait4 + 10
libsystem_kernel.dylib`__wait4:
->  0x7fff8f69e582 <+10>: jae    0x7fff8f69e58c            ; <+20>
    0x7fff8f69e584 <+12>: movq   %rax, %rdi
    0x7fff8f69e587 <+15>: jmp    0x7fff8f6987f2            ; cerror
    0x7fff8f69e58c <+20>: retq

Executable module set to "/usr/local/Cellar/mono/4.6.0.245/bin/mono".
Architecture set to: x86_64h-apple-macosx.
(lldb) thread list
Process 52756 stopped
* thread #1: tid = 0xfa760, 0x00007fff8f69e582 libsystem_kernel.dylib`__wait4 + 10, name = 'tid_a0b', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  thread #2: tid = 0xfa762, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'SGen worker'
  thread #3: tid = 0xfa763, 0x00007fff8f697fae libsystem_kernel.dylib`semaphore_wait_trap + 10, name = 'Finalizer'
  thread #4: tid = 0xfa764, 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
  thread #5: tid = 0xfa765, 0x00007fff8f69eefa libsystem_kernel.dylib`kevent_qos + 10, queue = 'com.apple.libdispatch-manager'
  thread #6: tid = 0xfa766, 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
  thread #7: tid = 0xfa778, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'Timer-Scheduler'
  thread #8: tid = 0xfa779, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'tid_250f'
  thread #9: tid = 0xfa77a, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'Threadpool worker'
  thread #10: tid = 0xfa77b, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'Threadpool worker'
  thread #11: tid = 0xfa77c, 0x00007fff8f69e07a libsystem_kernel.dylib`__select + 10, name = 'tid_360b'
  thread #12: tid = 0xfa77d, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'Threadpool worker'
  thread #13: tid = 0xfa859, 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
  thread #14: tid = 0xfa85a, 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
(lldb) thread backtrace all
* thread #1: tid = 0xfa760, 0x00007fff8f69e582 libsystem_kernel.dylib`__wait4 + 10, name = 'tid_a0b', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  * frame #0: 0x00007fff8f69e582 libsystem_kernel.dylib`__wait4 + 10
    frame #1: 0x000000010a2cc7e0 mono`mono_handle_native_sigsegv + 421
    frame #2: 0x000000010a31c8dc mono`mono_arch_handle_altstack_exception + 90
    frame #3: 0x000000010a2330f9 mono`mono_sigsegv_signal_handler + 440
    frame #4: 0x00007fff97e8d52a libsystem_platform.dylib`_sigtramp + 26
    frame #5: 0x00007fff9686d892 CoreGraphics`CGDisplayBounds + 48
    frame #6: 0x0000000112b88fdf
    frame #7: 0x0000000112b8294f
    frame #8: 0x000000010a235eee mono`mono_jit_runtime_invoke + 2180
    frame #9: 0x000000010a3c7fb8 mono`do_runtime_invoke + 88
    frame #10: 0x000000010a3c4983 mono`mono_runtime_class_init_full + 648
    frame #11: 0x000000010a2304b6 mono`mono_jit_compile_method_inner + 2269
    frame #12: 0x000000010a232908 mono`mono_jit_compile_method_with_opt + 608
    frame #13: 0x000000010a2ceb29 mono`common_call_trampoline + 1135
    frame #14: 0x000000010a2ce68c mono`mono_magic_trampoline + 73
    frame #15: 0x000000010a804298
    frame #16: 0x0000000112b823f9
    frame #17: 0x000000010a235eee mono`mono_jit_runtime_invoke + 2180
    frame #18: 0x000000010a3c7fb8 mono`do_runtime_invoke + 88
    frame #19: 0x000000010a3cae0a mono`mono_runtime_try_invoke_array + 1232
    frame #20: 0x000000010a34ed5b mono`ves_icall_InternalInvoke + 691
    frame #21: 0x000000010e9f68c4
    frame #22: 0x000000010c4d1f66 mscorlib.dll.dylib`System_RuntimeType_InvokeMember_string_System_Reflection_BindingFlags_System_Reflection_Binder_object_object___System_Reflection_ParameterModifier___System_Globalization_CultureInfo_string__ + 3766
    frame #23: 0x000000010c659c9a mscorlib.dll.dylib`System_Reflection_Emit_TypeBuilder_InvokeMember_string_System_Reflection_BindingFlags_System_Reflection_Binder_object_object___System_Reflection_ParameterModifier___System_Globalization_CultureInfo_string__ + 106
    frame #24: 0x000000010c5b5ccf mscorlib.dll.dylib`System_Type_InvokeMember_string_System_Reflection_BindingFlags_System_Reflection_Binder_object_object___System_Globalization_CultureInfo + 95
    frame #25: 0x0000000112ac06c9
    frame #26: 0x0000000112ac04f3
    frame #27: 0x0000000112136572
    frame #28: 0x000000011213562f
    frame #29: 0x00000001114b215a
    frame #30: 0x00000001121339b9
    frame #31: 0x0000000112130d87
    frame #32: 0x0000000112130ba5
    frame #33: 0x00000001114b4846
    frame #34: 0x00000001114b23fb
    frame #35: 0x00000001114b1f4b

  thread #2: tid = 0xfa762, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'SGen worker'
    frame #0: 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fff8e9f9728 libsystem_pthread.dylib`_pthread_cond_wait + 767
    frame #2: 0x000000010a41e5a9 mono`thread_func + 312
    frame #3: 0x00007fff8e9f899d libsystem_pthread.dylib`_pthread_body + 131
    frame #4: 0x00007fff8e9f891a libsystem_pthread.dylib`_pthread_start + 168
    frame #5: 0x00007fff8e9f6351 libsystem_pthread.dylib`thread_start + 13

  thread #3: tid = 0xfa763, 0x00007fff8f697fae libsystem_kernel.dylib`semaphore_wait_trap + 10, name = 'Finalizer'
    frame #0: 0x00007fff8f697fae libsystem_kernel.dylib`semaphore_wait_trap + 10
    frame #1: 0x000000010a3c2419 mono`finalizer_thread + 189
    frame #2: 0x000000010a3a2f81 mono`start_wrapper + 420
    frame #3: 0x000000010a44693f mono`inner_start_thread + 383
    frame #4: 0x00007fff8e9f899d libsystem_pthread.dylib`_pthread_body + 131
    frame #5: 0x00007fff8e9f891a libsystem_pthread.dylib`_pthread_start + 168
    frame #6: 0x00007fff8e9f6351 libsystem_pthread.dylib`thread_start + 13

  thread #4: tid = 0xfa764, 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #0: 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x00007fff8e9f8578 libsystem_pthread.dylib`_pthread_wqthread + 1283
    frame #2: 0x00007fff8e9f6341 libsystem_pthread.dylib`start_wqthread + 13

  thread #5: tid = 0xfa765, 0x00007fff8f69eefa libsystem_kernel.dylib`kevent_qos + 10, queue = 'com.apple.libdispatch-manager'
    frame #0: 0x00007fff8f69eefa libsystem_kernel.dylib`kevent_qos + 10
    frame #1: 0x00007fff8e9c6165 libdispatch.dylib`_dispatch_mgr_invoke + 216
    frame #2: 0x00007fff8e9c5dcd libdispatch.dylib`_dispatch_mgr_thread + 52

  thread #6: tid = 0xfa766, 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #0: 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x00007fff8e9f8578 libsystem_pthread.dylib`_pthread_wqthread + 1283
    frame #2: 0x00007fff8e9f6341 libsystem_pthread.dylib`start_wqthread + 13

  thread #7: tid = 0xfa778, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'Timer-Scheduler'
    frame #0: 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fff8e9f9728 libsystem_pthread.dylib`_pthread_cond_wait + 767
    frame #2: 0x000000010a423e8c mono`mono_os_cond_timedwait + 131
    frame #3: 0x000000010a423c6c mono`_wapi_handle_timedwait_signal_handle + 527
    frame #4: 0x000000010a434564 mono`wapi_WaitForSingleObjectEx + 618
    frame #5: 0x000000010a39f6b1 mono`mono_wait_uninterrupted + 109
    frame #6: 0x000000010a39f919 mono`ves_icall_System_Threading_WaitHandle_WaitOne_internal + 104
    frame #7: 0x0000000112b16f9f
    frame #8: 0x000000010c5a2120 mscorlib.dll.dylib`System_Threading_WaitHandle_InternalWaitOne_System_Runtime_InteropServices_SafeHandle_long_bool_bool + 64
    frame #9: 0x000000010c5a20cc mscorlib.dll.dylib`System_Threading_WaitHandle_WaitOne_long_bool + 44
    frame #10: 0x000000010c5a1f33 mscorlib.dll.dylib`System_Threading_WaitHandle_WaitOne_int_bool + 51
    frame #11: 0x000000010c5a205c mscorlib.dll.dylib`System_Threading_WaitHandle_WaitOne_int + 28
    frame #12: 0x000000010c711828 mscorlib.dll.dylib`System_Threading_Timer_Scheduler_SchedulerThread + 2552
    frame #13: 0x000000010c59ba8b mscorlib.dll.dylib`System_Threading_ThreadHelper_ThreadStart_Context_object + 155
    frame #14: 0x000000010c599671 mscorlib.dll.dylib`System_Threading_ExecutionContext_Run_System_Threading_ExecutionContext_System_Threading_ContextCallback_object_bool + 33
    frame #15: 0x000000010c5995fa mscorlib.dll.dylib`System_Threading_ExecutionContext_Run_System_Threading_ExecutionContext_System_Threading_ContextCallback_object + 90
    frame #16: 0x000000010c59bbdf mscorlib.dll.dylib`System_Threading_ThreadHelper_ThreadStart + 47
    frame #17: 0x000000010a9332f1
    frame #18: 0x000000010a235eee mono`mono_jit_runtime_invoke + 2180
    frame #19: 0x000000010a3c7fb8 mono`do_runtime_invoke + 88
    frame #20: 0x000000010a3a2fb1 mono`start_wrapper + 468
    frame #21: 0x000000010a44693f mono`inner_start_thread + 383
    frame #22: 0x00007fff8e9f899d libsystem_pthread.dylib`_pthread_body + 131
    frame #23: 0x00007fff8e9f891a libsystem_pthread.dylib`_pthread_start + 168
    frame #24: 0x00007fff8e9f6351 libsystem_pthread.dylib`thread_start + 13

  thread #8: tid = 0xfa779, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'tid_250f'
    frame #0: 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fff8e9f9728 libsystem_pthread.dylib`_pthread_cond_wait + 767
    frame #2: 0x000000010a444ef3 mono`mono_thread_info_sleep + 626
    frame #3: 0x000000010a3a5aa6 mono`monitor_thread + 149
    frame #4: 0x000000010a3a2f81 mono`start_wrapper + 420
    frame #5: 0x000000010a44693f mono`inner_start_thread + 383
    frame #6: 0x00007fff8e9f899d libsystem_pthread.dylib`_pthread_body + 131
    frame #7: 0x00007fff8e9f891a libsystem_pthread.dylib`_pthread_start + 168
    frame #8: 0x00007fff8e9f6351 libsystem_pthread.dylib`thread_start + 13

  thread #9: tid = 0xfa77a, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'Threadpool worker'
    frame #0: 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fff8e9f9728 libsystem_pthread.dylib`_pthread_cond_wait + 767
    frame #2: 0x000000010a3a63e9 mono`worker_thread + 1232
    frame #3: 0x000000010a3a2f81 mono`start_wrapper + 420
    frame #4: 0x000000010a44693f mono`inner_start_thread + 383
    frame #5: 0x00007fff8e9f899d libsystem_pthread.dylib`_pthread_body + 131
    frame #6: 0x00007fff8e9f891a libsystem_pthread.dylib`_pthread_start + 168
    frame #7: 0x00007fff8e9f6351 libsystem_pthread.dylib`thread_start + 13

  thread #10: tid = 0xfa77b, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'Threadpool worker'
    frame #0: 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fff8e9f9728 libsystem_pthread.dylib`_pthread_cond_wait + 767
    frame #2: 0x000000010a3a63e9 mono`worker_thread + 1232
    frame #3: 0x000000010a3a2f81 mono`start_wrapper + 420
    frame #4: 0x000000010a44693f mono`inner_start_thread + 383
    frame #5: 0x00007fff8e9f899d libsystem_pthread.dylib`_pthread_body + 131
    frame #6: 0x00007fff8e9f891a libsystem_pthread.dylib`_pthread_start + 168
    frame #7: 0x00007fff8e9f6351 libsystem_pthread.dylib`thread_start + 13

  thread #11: tid = 0xfa77c, 0x00007fff8f69e07a libsystem_kernel.dylib`__select + 10, name = 'tid_360b'
    frame #0: 0x00007fff8f69e07a libsystem_kernel.dylib`__select + 10
    frame #1: 0x000000010a43ee2c mono`mono_poll + 399
    frame #2: 0x000000010a3a7d80 mono`poll_event_wait + 113
    frame #3: 0x000000010a3a78ef mono`selector_thread + 1104
    frame #4: 0x000000010a3a2f81 mono`start_wrapper + 420
    frame #5: 0x000000010a44693f mono`inner_start_thread + 383
    frame #6: 0x00007fff8e9f899d libsystem_pthread.dylib`_pthread_body + 131
    frame #7: 0x00007fff8e9f891a libsystem_pthread.dylib`_pthread_start + 168
    frame #8: 0x00007fff8e9f6351 libsystem_pthread.dylib`thread_start + 13

  thread #12: tid = 0xfa77d, 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10, name = 'Threadpool worker'
    frame #0: 0x00007fff8f69ddb6 libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fff8e9f9728 libsystem_pthread.dylib`_pthread_cond_wait + 767
    frame #2: 0x000000010a3a63e9 mono`worker_thread + 1232
    frame #3: 0x000000010a3a2f81 mono`start_wrapper + 420
    frame #4: 0x000000010a44693f mono`inner_start_thread + 383
    frame #5: 0x00007fff8e9f899d libsystem_pthread.dylib`_pthread_body + 131
    frame #6: 0x00007fff8e9f891a libsystem_pthread.dylib`_pthread_start + 168
    frame #7: 0x00007fff8e9f6351 libsystem_pthread.dylib`thread_start + 13

  thread #13: tid = 0xfa859, 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #0: 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x00007fff8e9f8578 libsystem_pthread.dylib`_pthread_wqthread + 1283
    frame #2: 0x00007fff8e9f6341 libsystem_pthread.dylib`start_wqthread + 13

  thread #14: tid = 0xfa85a, 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #0: 0x00007fff8f69e5e2 libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x00007fff8e9f8578 libsystem_pthread.dylib`_pthread_wqthread + 1283
    frame #2: 0x00007fff8e9f6341 libsystem_pthread.dylib`start_wqthread + 13
(lldb) detach

=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries
used by your application.
=================================================================

Process 52756 detached
(lldb) ./run.sh: line 4: 52756 Abort trap: 6           fsharpi src/scrape.fsx
 ✘  ~/code/fsharp-lang-suggestions   canopy_test_osx ●  (lldb) quit
```
It seems link 64 bit chrome driver is not a supported configuration, and fsharpi runs at 64-bit by default. I haven't yet tried finding/building fsharpi for 32-bits to see if that works.

**Comment by lefthandedgoat** [comment]
I think the problem is when it tries to launch the browser, it tries to pin it to the right half of the screen.  This is configurable and can be turned off.

```
  at System.Windows.Forms.Screen..cctor () <0x0011f>
  at (wrapper runtime-invoke) object.runtime_invoke_void (object,intptr,intptr,intptr) <0x0006e>
  at <unknown> <0xffffffff>
  at canopy.core.pin (canopy.types/direction) <0x00028>
```

Try this before starting the browser.

```
canopy.configuration. autoPinBrowserRightOnLaunch <- false
```

**Comment by baronfel** [comment]
yup, I tried that in my script and it behaves in the same way.  I can't even get the safari driver to run, so no cross-browser test as of yet.  Is there anything I should be looking for to provide more/better info?

**Comment by lefthandedgoat** [comment]
Can you clone this and run `sh build.sh`

https://github.com/lefthandedgoat/canopyStarterKit

I am on mono 4.4.0 and it works for me.  I am curious if its newer mono or something else.

**Comment by lefthandedgoat** [comment]
Also, clone your project, switched branches, and it runs for me.

```
iver/chromedriver.exe" to "/Users/cholt1/projects/fsharp-lang-suggestions/src/bin/Release/chromedriver.exe".
Done Building Project "/Users/cholt1/projects/fsharp-lang-suggestions/src/fslang-migration.fsproj" (default targets).
Done Building Project "/Users/cholt1/projects/fsharp-lang-suggestions/fslang-migration.sln" (default targets).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:05.49
Starting ChromeDriver 2.21.371459 (36d3d07f660ff2bc1bf28a75d1cdabed0983e7c4) on port 59411
Only local connections are allowed.
....
....
https://code.google.com/p/chromedriver/issues/detail?id=1249 for details and workarounds.
found 492 ideas
parsing idea 0: https://fslang.uservoice.com/forums/245727-f-language/suggestions/16529041-close-user-voice-in-favor-of-github-issues
```

**Comment by baronfel** [comment]
that's great! your repo works flawlessly for me, by the way. interesting that yours is a compiled executable vs a script being run.  let me try making mine an executable and see if that addresses the issue.
